maestro/scripts/adapt-device.py

```
@@ -0,0 +1,347 @@
+def save_profiles(path: Path, data: dict[str, Any]) -> None:
+    if yaml is None:
+        sys.exit("error: PyYAML required — pip install pyyaml")
+    path.write_text(yaml.safe_dump(data, sort_keys=False, default_flow_style=False))
+
+
+def adb_shell(device: str | None, *args: str) -> str:
+    cmd = ["adb"]
+    if device:
+        cmd.extend(["-s", device])
+    cmd.extend(["shell", *args])
+    result = subprocess.run(cmd, capture_output=True, text=True, check=False)
```

Ruff (0.15.15)

[error] `subprocess` call: check for execution of untrusted input

(S603)